crates/router/src/core/refunds.rs

```diff
@@ -284,14 +288,52 @@ pub async fn trigger_refund_to_gateway(
     };
 
     let refund_update = match router_data_res.response {
-        Err(err) => storage::RefundUpdate::ErrorUpdate {
-            refund_status: Some(enums::RefundStatus::Failure),
-            refund_error_message: err.reason.or(Some(err.message)),
-            refund_error_code: Some(err.code),
-            updated_by: storage_scheme.to_string(),
-            connector_refund_id: None,
-            connector_refund_data: None,
-        },
+        Err(err) => {
+            let option_gsm = helpers::get_gsm_record(
+                state,
+                Some(err.code.clone()),
+                Some(err.message.clone()),
+                connector.connector_name.to_string(),
+                consts::REFUND_FLOW_STR.to_string(),
+            )
+            .await;
+
+            let gsm_unified_code = option_gsm.as_ref().and_then(|gsm| gsm.unified_code.clone());
+            let gsm_unified_message = option_gsm.and_then(|gsm| gsm.unified_message);
+
+            let (unified_code, unified_message) = if let Some((code, message)) =
+                gsm_unified_code.as_ref().zip(gsm_unified_message.as_ref())
+            {
+                (code.to_owned(), message.to_owned())
+            } else {
+                (
+                    consts::DEFAULT_UNIFIED_ERROR_CODE.to_owned(),
+                    consts::DEFAULT_UNIFIED_ERROR_MESSAGE.to_owned(),
+                )
+            };
+            let header_map = &state.get_request_headers();
+            let locale = utils::get_locale_from_header(header_map.as_ref());
+
+            let unified_translated_message = helpers::get_unified_translation(
+                state,
+                unified_code.to_owned(),
+                unified_message.to_owned(),
+                locale.to_owned(),
+            )
+            .await
+            .or(Some(unified_message));
+
+            storage::RefundUpdate::ErrorUpdate {
+                refund_status: Some(enums::RefundStatus::Failure),
+                refund_error_message: err.reason.or(Some(err.message)),
+                refund_error_code: Some(err.code),
+                updated_by: storage_scheme.to_string(),
+                connector_refund_id: None,
+                connector_refund_data: None,
+                unified_code: Some(unified_code),
+                unified_message: unified_translated_message,
```

Comment:
This should always be in English when storing in DB

```suggestion
                unified_message: Some(unified_message),
```